### PR TITLE
[IE][VPU][NGraph] Move dynamic NMS from common to vpu folder

### PR DESCRIPTION
**Pull request by AndrewBakalinIntel**
Ticket - #-34827
Changes:
- Move dynamic NMS and appropriate transformation to vpu folder
- Replace v3::NMS base class with v4::NMS base class with has correct shape infer function
- Use ::vpu in error.hpp to avoid namespace clashing.

### CODE REVIEW

File: inference-engine/src/vpu/common/include/vpu/utils/error.hpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ void throwFormat(const char* fileName, int lineNumber, const char* messageFormat
 #define VPU_THROW_UNLESS(condition, ...)                                                       \
     do {                                                                                       \
         if (!(condition)) {                                                                    \
-            vpu::details::throwFormat<vpu::details::VPUException>(__FILE__, __LINE__, __VA_ARGS__); \
+            ::vpu::details::throwFormat<::vpu::details::VPUException>(__FILE__, __LINE__, __VA_ARGS__); \
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/common/include/vpu/utils/error.hpp:44]: Could you please explain why you need this?

Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/common/include/vpu/utils/error.hpp:44]: When I tried to add dynamicNMS transformation I've found namespace clashing between `vpu` and `ngraph::vpu` namespaces, so here I resolve it.

**Comment by AndrewBakalinIntel** [comment]
@GlebKazantaev @jane-intel @lazarevevgeny could you please take a look?

**Comment by AndrewBakalinIntel** [comment]
@Maxim-Doronin could you please merge it?

**Pull request merged**